ush/forecast_postdet.sh

```diff
@@ -695,6 +695,7 @@ CMEPS_postdet() {
       else
         restart_date="${current_cycle}"
       fi
+      RUN=${rCDUMP} YMD=${restart_date:0:8} HH=${restart_date:8:2} declare_from_tmpl COM_MED_RESTART_PREV:COM_MED_RESTART_TMPL
```

Comment:
@aerorahul I know this was not originally the way that you wanted to fix issue #2528 however, the gdas forecast succeeds when I add this.  

Here's the successful log: /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test02/COMROOT/c48t02/logs/2021032412/gdasfcst.log

I'll start comparing that with a failed log: /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test01/COMROOT/c48t01/logs/2021032412/gdasfcst.log.0 

to see if we can learn why this line is needed?

Reply:
Okay so from the failed log we get: 

```
+ JGLOBAL_FORECAST[53]: declare_from_tmpl -rx COM_OCEAN_RESTART_PREV:COM_OCEAN_RESTART_TMPL COM_MED_RESTART_PREV:COM_MED_RESTART_TMPL
+ bash_utils.sh[35]: [[ NO == \N\O ]]
+ bash_utils.sh[35]: set +x
declare_from_tmpl :: COM_OCEAN_RESTART_PREV=/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test01/COMROOT/c48t01/gdas.20210324/06//model_data/ocean/restart
declare_from_tmpl :: COM_MED_RESTART_PREV=/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test01/COMROOT/c48t01/gdas.20210324/06//model_data/med/restart
```

and then lower down... 

```
+ forecast_postdet.sh[681]: echo 'SUB CMEPS_postdet: Linking output data for CMEPS mediator'
SUB CMEPS_postdet: Linking output data for CMEPS mediator
+ forecast_postdet.sh[683]: [[ .true. == \.\t\r\u\e\. ]]
+ forecast_postdet.sh[686]: local restart_date cmeps_restart_file
+ forecast_postdet.sh[687]: [[ NO == \Y\E\S ]]
+ forecast_postdet.sh[693]: [[ YES == \Y\E\S ]]
+ forecast_postdet.sh[694]: restart_date=2021032409
+ forecast_postdet.sh[698]: cmeps_restart_file=/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test01/COMROOT/c48t01/gdas.20210324/06//model_data/med/restart/20210324.090000.ufs.cpld.cpl.r.nc
+ forecast_postdet.sh[702]: [[ -f /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test01/COMROOT/c48t01/gdas.20210324/06//model_data/med/restart/20210324.090000.ufs.cpld.cpl.r.nc ]]
+ forecast_postdet.sh[703]: /bin/cp -p /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test01/COMROOT/c48t01/gdas.20210324/06//model_data/med/restart/20210324.090000.ufs.cpld.cpl.r.nc /scratch1/NCEPDEV/stmp2/Jessica.Meixner/RUNDIRS/c48t01/gdasfcst.2021032412/fcst.932707/ufs.cpld.cpl.r.nc
+ forecast_postdet.sh[705]: rm -f /scratch1/NCEPDEV/stmp2/Jessica.Meixner/RUNDIRS/c48t01/gdasfcst.2021032412/fcst.932707/rpointer.cpl
+ forecast_postdet.sh[706]: touch /scratch1/NCEPDEV/stmp2/Jessica.Meixner/RUNDIRS/c48t01/gdasfcst.2021032412/fcst.932707/rpointer.cpl
+ forecast_postdet.sh[707]: echo ufs.cpld.cpl.r.nc
```

and then we fail: 

```
 0:  zeroing coupling accumulated fields at kdt=           10
 0:  zeroing coupling accumulated fields at kdt=           10
 0: -------->(med_phases_restart_read) mediating for: 2021  3 24  9  0  0   0
 0: Abort(1) on node 0 (rank 0 in comm 496): application called MPI_Abort(comm=0x84000003, 1) - process 0
 4: Abort(1) on node 4 (rank 4 in comm 496): application called MPI_Abort(comm=0x84000002, 1) - process 4
 3: Abort(1) on node 3 (rank 3 in comm 496): application called MPI_Abort(comm=0x84000002, 1) - process 3
 5: Abort(1) on node 5 (rank 5 in comm 496): application called MPI_Abort(comm=0x84000002, 1) - process 5
 2: Abort(1) on node 2 (rank 2 in comm 496): application called MPI_Abort(comm=0x84000002, 1) - process 2
 1: Abort(1) on node 1 (rank 1 in comm 496): application called MPI_Abort(comm=0x84000002, 1) - process 1
srun: error: h8c48: tasks 0-39: Exited with exit code 1
```


And from the successful log we get: 

```
+ JGLOBAL_FORECAST[53]: declare_from_tmpl -rx COM_OCEAN_RESTART_PREV:COM_OCEAN_RESTART_TMPL COM_MED_RESTART_PREV:COM_MED_RESTART_TMPL
+ bash_utils.sh[35]: [[ NO == \N\O ]]
+ bash_utils.sh[35]: set +x
declare_from_tmpl :: COM_OCEAN_RESTART_PREV=/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test02/COMROOT/c48t02/gdas.20210324/06//model_data/ocean/restart
declare_from_tmpl :: COM_MED_RESTART_PREV=/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test02/COMROOT/c48t02/gdas.20210324/06//model_data/med/restart
```


```
+ forecast_postdet.sh[698]: declare_from_tmpl COM_MED_RESTART_PREV:COM_MED_RESTART_TMPL
+ bash_utils.sh[35]: [[ NO == \N\O ]]
+ bash_utils.sh[35]: set +x
declare_from_tmpl :: COM_MED_RESTART_PREV=/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test02/COMROOT/c48t02/gdas.20210324/09//model_data/med/restart
+ forecast_postdet.sh[699]: cmeps_restart_file=/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test02/COMROOT/c48t02/gdas.20210324/09//model_data/med/restart/20210324.090000.ufs.cpld.cpl.r.nc
+ forecast_postdet.sh[703]: [[ -f /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test02/COMROOT/c48t02/gdas.20210324/09//model_data/med/restart/20210324.090000.ufs.cpld.cpl.r.nc ]]
+ forecast_postdet.sh[713]: echo 'WARNING: CMEPS restart file '\''/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test02/COMROOT/c48t02/gdas.20210324/09//model_data/med/restart/20210324.090000.ufs.cpld.cpl.r.nc'\'' not found for warm_start='\''.true.'\'', will initialize!'
WARNING: CMEPS restart file '/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/iau_03/test02/COMROOT/c48t02/gdas.20210324/09//model_data/med/restart/20210324.090000.ufs.cpld.cpl.r.nc' not found for warm_start='.true.', will initialize!

```


So I guess this is the next question is, which is the correct mediator restart file? And if the "failed" case is the correct restart file, and it does appear to exist, then why isn't the model failing?

Reply:
Okay,, I just re-read the successful run's COM_MED_RESTART_PREV since we do not have 09 cycle's so this is working for the wrong reasons and this is not the correct fix!

Reply:
Could  it be that the mediator restart that we have for this test is wrong and that we've been "boot strapping" it without noticing? Why not just erasing the file and let the model initialize the mediator as in your second test above?

Reply:
@guillaumevernieres that might be the case.  I went and looked at a log file from another successful run but before the change for restarts and found: 
```
+ forecast_postdet.sh[980]: local mediator_file=/scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/turnonmarine/c48ci01/COMROOT/c48t01/gdas.20210324/12//model_data/med/restart/20210324.120000.ufs.cpld.cpl.r.nc
+ forecast_postdet.sh[981]: [[ -f /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/turnonmarine/c48ci01/COMROOT/c48t01/gdas.20210324/12//model_data/med/restart/20210324.120000.ufs.cpld.cpl.r.nc ]]
+ forecast_postdet.sh[990]: echo 'WARNING: /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/turnonmarine/c48ci01/COMROOT/c48t01/gdas.20210324/12//model_data/med/restart/20210324.120000.ufs.cpld.cpl.r.nc does not exist for warm_start = .true., initializing!'
WARNING: /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/turnonmarine/c48ci01/COMROOT/c48t01/gdas.20210324/12//model_data/med/restart/20210324.120000.ufs.cpld.cpl.r.nc does not exist for warm_start = .true., initializing!
+ exglobal_forecast.sh[125]: [[ .true. = .true. ]]
```
That we are not using the mediator restart file... and were looking for the wrong file.  So it's very possible that we needed an updated restart because of updates from a while ago in the model and just never noticed.  This log file is here: /scratch1/NCEPDEV/climate/Jessica.Meixner/cycling/turnonmarine/c48ci01/COMROOT/c48t01/logs/2021032412/gdasfcst.log

I've looked more at the dates and such and I do think the failed run above without this line was looking for the correct cmeps restart file.  It's very possible that the issue here is that the cplr restart file changed in a previous model version (perhaps a while ago) and we just didn't realize it wasn't correct because we were never looking for the correct file and were starting without it.  

I can try a test run deleting this file to see if things run for this test and report back.  Then, we can decide if we should just remove it or recreate a correct coupler file.